src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -711,48 +703,29 @@ private async Task ComputeAnalyzerDiagnosticsAsync(AnalysisScope? analysisScope,
             {
                 throw ExceptionUtilities.Unreachable();
             }
-            finally
-            {
-                if (driverAllocatedFromPool)
-                    FreeDriver(driver);
-            }
 
```

Comment:
nit: add a return-statement here to make it clear that there's no more code after this point except nested functions.